hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecord.java

```diff
@@ -90,7 +91,11 @@ public HoodieOperation getHoodieOperation() {
 
   public BufferedRecord<T> toBinary(RecordContext<T> recordContext) {
     if (record != null) {
-      record = recordContext.seal(recordContext.toBinaryRow(recordContext.getSchemaFromBufferRecord(this), record));
+      HoodieSchema schema = recordContext.getSchemaFromBufferRecord(this);
+      // Schema can be null in test scenarios where schemas are not registered in the RecordContext (e.g. in tests)
+      if (schema != null) {
+        record = recordContext.seal(recordContext.toBinaryRow(schema, record));
```

Comment:
Do we still want to `seal` in these cases?

Reply:
This is in the original code. AM just adding an additional check to ensure that schema is not null. 

`org.apache.hudi.common.engine.RecordContext#getSchemaFromBufferRecord` calls
`org.apache.hudi.common.engine.RecordContext#decodeAvroSchema`, which is a nullable method.